railties/test/generators/shared_generator_tests.rb

```diff
@@ -91,8 +104,10 @@ def test_template_is_executed_when_supplied_an_https_path
     template = %{ say "It works!" }
     template.instance_eval "def read; self; end" # Make the string respond to read
 
-    generator([destination_root], template: path).expects(:open).with(path, 'Accept' => 'application/x-thor-template').returns(template)
-    quietly { assert_match(/It works!/, capture(:stdout) { generator.invoke_all }) }
+    generator = generator([destination_root], template: path)
+    assert_called_with(generator, :open, [path, 'Accept' => 'application/x-thor-template'], returns: template) do
```

Comment:
@kaspth Can you please also help me with this test? Getting following error with this.

````bash
  1) Error:
AppGeneratorTest#test_template_is_executed_when_supplied_an_https_path:
NoMethodError: undefined method `read' for nil:NilClass
    /home/travis/build/rails/rails/vendor/bundle/ruby/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:218:in `block in apply'
````
I tried to reproduced this error in separate test. following test runs properly with only `stub` but gives same error with `assert_called_with` and `test_template_is_executed_when_supplied_an_https_path` test gives error in both cases
````ruby 
class Temp
  def foo; end
end

def test_string_template
  template = "test"
  template.instance_eval "def read; self; end" # Make the string respond to read

  temp = Temp.new
  assert_called_with(temp, :foo, returns: template) do
  # temp.stub(:foo, template) do
    temp.foo {|io| io.read }
  end
end
````